vms/platformvm/vm.go

```diff
@@ -347,6 +443,10 @@ func (vm *VM) SetState(_ context.Context, state snow.State) error {
 
 // Shutdown this blockchain
 func (vm *VM) Shutdown(context.Context) error {
+	if vm.cancel != nil {
+		vm.cancel()
+	}
```

Comment:
If we move this after `vm.db == nil` then I don't think we need the `nil` check

Reply:
Moved